src/libraries/System.Runtime/tests/System/DoubleTests.cs

```diff
@@ -272,6 +272,25 @@ public static IEnumerable<object[]> Parse_Valid_TestData()
             yield return new object[] { (567.89).ToString(), defaultStyle, null, 567.89 };
             yield return new object[] { (-567.89).ToString(), defaultStyle, null, -567.89 };
             yield return new object[] { "1E23", defaultStyle, null, 1E23 };
+            yield return new object[] { "9007199254740997.0", defaultStyle, invariantFormat, 9007199254740996.0 };
+            yield return new object[] { "9007199254740997.0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000", defaultStyle, invariantFormat, 9007199254740996.0 };
```

Comment:
It looks like we are missing a zero in most of these cases.

The longest double is `767` significant digits, which is how many digits exist in this string. The buffer then has space for the rounding digit and the null terminator (769 digits).

Reply:
I actually added tests for string of length `maxDigCount` + 1/2/3 to test various scenarios now.

Reply:
Might be easier to read as something like `"9007199254740997." + new string('0', 751)` etc

Reply:
Ugh, I should've thought of that :@ Will change this on my next pass over this file.